gdal/frmts/tiledb/tiledbdataset.cpp

```diff
@@ -1135,6 +1262,45 @@ GDALDataset *TileDBDataset::Open( GDALOpenInfo * poOpenInfo )
 
         tiledb::ArraySchema schema = poDS->m_array->schema();
 
+        char ** papszStructMeta = poDS->GetMetadata( "IMAGE_STRUCTURE" );
+        const char* pszXSize = CSLFetchNameValue( papszStructMeta, "X_SIZE");
+        if ( pszXSize )
+        {
+            poDS->nRasterXSize = atoi( pszXSize );
+        }
+
+        const char* pszYSize = CSLFetchNameValue( papszStructMeta, "Y_SIZE");
+        if ( pszYSize )
+        {
+            poDS->nRasterYSize = atoi( pszYSize );
+        }
+
+        const char* pszNBits = CSLFetchNameValue( papszStructMeta, "NBITS");
+        if ( pszNBits )
+        {
+            poDS->nBitsPerSample = atoi( pszNBits );
+        }
+
+        const char* pszDataType = CSLFetchNameValue( papszStructMeta, "DATA_TYPE");
+        if ( pszDataType )
+        {
+            // handle the case where arrays have been written with int type (2.5.0)
+            GDALDataType eDT = GDALGetDataTypeByName( pszDataType );
+            if ( eDT == GDT_Unknown )
+            {
+                poDS->eDataType = static_cast<GDALDataType>( atoi( pszDataType ) );
```

Comment:
before casting to GDALDataType, it would be safer to check that the integer values are in the range of the enumeration (and different from GDT_Unknown)